updater/lib/dependabot/job.rb

```diff
@@ -43,87 +45,132 @@ class Job
       dependency_groups
       dependency_group_to_refresh
       repo_private
-    ).freeze
-
-    attr_reader :allowed_updates,
-                :credentials,
-                :dependencies,
-                :existing_pull_requests,
-                :existing_group_pull_requests,
-                :id,
-                :ignore_conditions,
-                :package_manager,
-                :requirements_update_strategy,
-                :security_advisories,
-                :security_updates_only,
-                :source,
-                :token,
-                :vendor_dependencies,
-                :dependency_groups,
-                :dependency_group_to_refresh
+    ).freeze, T::Array[Symbol])
+
+    sig { returns(T::Array[T::Hash[String, T.untyped]]) }
+    attr_reader :allowed_updates
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :credentials
+
+    sig { returns(T.nilable(T::Array[String])) }
+    attr_reader :dependencies
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :existing_pull_requests
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :existing_group_pull_requests
+
+    sig { returns(T.any(Integer, String)) }
+    attr_reader :id
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :ignore_conditions
+
+    sig { returns(String) }
+    attr_reader :package_manager
+
+    sig { returns(T.nilable(String)) }
+    attr_reader :requirements_update_strategy
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :security_advisories
+
+    sig { returns(T::Boolean) }
+    attr_reader :security_updates_only
+
+    sig { returns(Dependabot::Source) }
+    attr_reader :source
+
+    sig { returns(T.nilable(String)) }
+    attr_reader :token
 
+    sig { returns(T::Boolean) }
+    attr_reader :vendor_dependencies
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :dependency_groups
+
+    sig { returns(T.nilable(String)) }
+    attr_reader :dependency_group_to_refresh
+
+    sig do
+      params(job_id: T.untyped, job_definition: T.untyped, repo_contents_path: T.untyped).returns(T.attached_class)
+    end
     def self.new_fetch_job(job_id:, job_definition:, repo_contents_path: nil)
-      attrs = standardise_keys(job_definition["job"]).slice(*PERMITTED_KEYS)
+      attrs = standardise_keys(job_definition["job"]).select { |k, _| PERMITTED_KEYS.include?(k) }
 
       new(attrs.merge(id: job_id, repo_contents_path: repo_contents_path))
     end
 
+    sig do
+      params(job_id: T.untyped, job_definition: T.untyped, repo_contents_path: T.untyped).returns(T.attached_class)
```

Comment:
`attached_class` is overkill here, unless you're planning to split this into a class hierarchy later?

Reply:
I don't remember doing that so probably once when I ran `bundle exec srb tc -a` it added it. That command can do some crazy things sometimes!